src/main/java/ProcessManager.java

```diff
@@ -0,0 +1,290 @@
+                    try {
+                        parseTasks(fileScan.nextLine());
+                    } catch (DukeException e) {
+                        e.printStatement();
+                    }
+                }
+                System.out.println("Task Successfully Imported\n");
+            } else {
+                file.getParentFile().mkdirs();
+                System.out.println("I am your very own schedule assistant here to enhanced your everyday life");
+                System.out.println("Type help to see what i can do!");
+            }
+        } catch (IOException e) {
+            System.out.println("Something went wrong during file creation :( ");
+        } catch (SecurityException e) {
+            System.out.println("File could not be accessed");
+        }
+    }
+
+    public void parseTasks(String line) throws DukeException {
+        int dividerPosition1 = line.indexOf("|") + 1;
+        int dividerPosition2 = line.indexOf("|", dividerPosition1) + 1;
+        int dividerPosition3 = line.indexOf("|", dividerPosition2) + 1;
+        String description = line.substring(dividerPosition1, dividerPosition2 - 1).trim();
+        String date = line.substring(dividerPosition2, dividerPosition3 - 1).trim();
+        String status = line.substring(dividerPosition3).trim();
+        if (line.startsWith("T")) {
+            taskList.add(new Todo(description));
+            try {
+                checkStatus(status);
+            } catch (DukeException e) {
+                e.printStatement();
+            }
+        } else if (line.startsWith("D")) {
+            taskList.add(new Deadline(description, date));
+            try {
+                checkStatus(status);
+            } catch (DukeException e) {
+                e.printStatement();
+            }
+        } else if (line.startsWith("E")) {
+            taskList.add(new Event(description, date));
+            try {
+                checkStatus(status);
+            } catch (DukeException e) {
+                e.printStatement();
+            }
+        } else {
+            throw new DukeException("Task Syntax Corrupted, Unable to Parse Request");
+        }
+    }
+
+    public void saveTasks() throws IOException {
+        FileWriter fileWrite = new FileWriter(FILEPATH);
+        fileWrite.close();
+        for (Task task : taskList) {
+            try {
+                task.saveTask(FILEPATH);
+            } catch (IOException e) {
+                throw new IOException("Error Occurred While Saving File");
+            }
+        }
+    }
+    /* ---- ------------- ---- */
+
+    /* ---- Handle Functions ---- */
+    public void handleEventRequest(String line) throws EventException {
+        if (!line.contains(EVENT_KEYWORD)) {
+            throw new EventException("Event Request Does Not Contain /at");
+        }
+        Integer dividerPosition = line.indexOf(EVENT_KEYWORD);
+        if (dividerPosition - TASK_DESCRIPTION_DIVIDER - EVENT_DIVIDER < 0) {
```

Comment:
Perhaps it would be better to put such operations in a variable to improve readability